### PR TITLE
Implement in-app tours

### PR DESCRIPTION
**Pull request by eecavanna**
In this branch, I implemented in-app tours. I used an NPM library called [reactour](https://github.com/elrumordelaluz/reactour) for this.

There is one "tour provider" for the entire app (you can think of this as a tour host, as you might encounter in a museum). In order to make it so different sections of the app (i.e. different rooms of the museum) appear to have section-specific tours; when the user goes to a given section of the app (that has such a tour available), that section of the app will tell the "tour provider" to disregard its current "itinerary" (i.e. `steps`) and use that section's itinerary (i.e. `steps`) instead. I implemented a custom React hook called `useLocalTour` to abstract away the details of that. 

📍 The hook is implemented within `components/CustomTourProvider/`, which may not be the most intuitive home for it.

A tour is a sequence of `steps`, where each step has two parts:
- A `selector`, which tells the tour provider which part of the page to highlight
- A `content` (i.e. a "piece of content"), which is the message or caption to display next to the highlighted thing

The caption appears in a `Popover`. In this branch, I've customized the `Popover` so that it lacks a "Back" arrow on the first step of a tour, and lacks a "Next" arrow on the final step of a tour. In the latter case, I added a button labeled "Dismiss," which ends that tour. 📍 Finally, I made some decisions about what the user can/can't do during the tour:

- The user can't exit the tour in the middle of the tour, unless they reload the web page
- When the tour is visible, the user can't interact with anything except the back/next/dots in the `Popover`

The tour content was inspired by the blurbs @jkelliher-github drafted in a Google Doc a few months ago. 📍 I _do_ expect to iterate on the tour content as part of this PR review process. For example, since the user can't interact with anything but the tour `Popover`, maybe I'll update the captions to be less imperative and more declarative; for example:

```diff
- Tap here to...
+ You can tap here to...
```

Props to @pkalita-lbl for suggesting the use of `useIonViewDidEnter` (as an alternative to `useEffect`) during today's squad meeting. That solved two problems (buggy behavior when resetting the tours; and unit tests that were failing due to calling a function outside of a provider).

Finally, I added a button to the "Settings" page, which users can use to reset the "tours seen" flags. If they press that button, the app will act like the user hasn't seen any of the tours yet.

Oops, I almost forgot to say: I also added a new browser storage key: `presentedTourIds`. This is a `Set` of the tour IDs of the tours that the app has already presented.

---

I used the 📍 emoji to highlight things that I can imagine us discussing in this PR (or on Slack).

### CODE REVIEW

**Comment by eecavanna** [comment]
Several tests are failing.

```js
 FAIL  src/components/StudyList/StudyList.test.tsx > StudyList should show study names
Error: checkWhetherTourHasBeenPresented called outside of provider
 ❯ checkWhetherTourHasBeenPresented src/Store.tsx:74:11
     72| 
     73|   checkWhetherTourHasBeenPresented: () => {
     74|     throw new Error(
       |           ^
     75|       "checkWhetherTourHasBeenPresented called outside of provider",
     76|     );
 ❯ src/components/CustomTourProvider/hooks.tsx:17:36

```

**Comment by eecavanna** [comment]
Here's a 1-minute video showing what the tour functionality looks like. For some reason, the video recorded didn't capture the newly-implemented toast notification that appears when the user pressed "Reset tours" on the "Settings" page.

https://github.com/user-attachments/assets/1570b4b6-c09b-4150-9209-295d2aabea9f

The animations still feel a little rough around the edges to me.

**Comment by eecavanna** [comment]
Here's a 1-minute video showing what the tour functionality looks like.

https://github.com/user-attachments/assets/8b1ffcbd-8c39-4922-95e0-332d0806700c

Not shown in this video: The toast notification that appears when the user resets the tours.

**Comment by eecavanna** [comment]
I cancelled the most recent GHA workflow run after seeing that it was still in the "Queued" state after 17+ minutes. I'll make another commit on this branch in order to trigger a new run.

**Review by pkalita-lbl — APPROVED** [review]
Overall I like it, and I agree with the decisions made here around the interactivity of highlighted elements. I just had one comment about the content of one tour step. 

My only other very non-urgent comment is about naming. My thinking is that `CustomTourProvider` isn't _just_ a customized `TourProvider`, it's _the_ `TourProvider` for our app. So I'd consider something like `AppTourProvider` for the component and `useAppTour` for the hook. Just throwing that out there; you can take it or leave it.

**Comment by eecavanna** [comment]
> My only other very non-urgent comment is about naming.

Thanks, @pkalita-lbl! I do prefer the names you suggested. I have updated this branch so it now uses them.

File: src/components/StudyList/StudyList.tsx [hunk 1]
```diff
@@ -18,8 +18,27 @@ import { getSubmissionSamples } from "../../utils";
...
+// Make steps for the tour.
+// Reference: https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes
+const steps: Array<StepType> = [
+  {
+    selector: `[data-tour="${TourId.StudyList}-1"]`,
+    content: "Tap here to create a new study.",
+  },
+  {
+    selector: `[data-tour="${TourId.StudyList}-2"]`,
+    content: "Tap here to edit an existing study.",
```
[the remark this answers is not in the document]
Comment (eecavanna) [reply src/components/StudyList/StudyList.tsx:34]: I agree. I will implement this (at least roughly).

[the remark this answers is not in the document]
Comment (eecavanna) [reply src/components/StudyList/StudyList.tsx]: I implemented this (at least roughly). Here's a screenshot of the result:

image

**Pull request merged**